python/ray/data/dataset.py

```diff
@@ -350,6 +356,12 @@ def map(
 
         transform_fn = generate_map_rows_fn()
 
+        if num_cpus is not None:
+            ray_remote_args["num_cpus"] = num_cpus
+
+        if num_gpus is not None:
+            ray_remote_args["num_gpus"] = num_gpus
+
```

Comment:
Should we raise a warning if both `num_cpus` and `ray_remote_args["num_cpus"]` are specified (and likewise for `num_gpus`? Not sure if it's worth adding the warning, but it's definitely an unexpected input

Reply:
I think this is not possible since ray_remote_args is accepts as **kwargs?

Reply:
kwargs desugars as a normal dictionary, so you can definitely check this. I think we should raise ValueError here actually, since it's clearly ambiguous?

Reply:
What @amogkam is saying this can't actually happen, since Python will already do the right thing, and I believe he is right:
```python
In [1]: def f(arg=None, **d):
   ...:     return (arg, d)
   ...: 

In [2]: f(arg=1, **{"arg": 2})
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Input In [2], in <cell line: 1>()
----> 1 f(arg=1, **{"arg": 2})

TypeError: f() got multiple values for keyword argument 'arg'

In [3]: f(**{"arg": 2})
Out[3]: (2, {})

In [4]: f(arg=1)
Out[4]: (1, {})
```